tools/doc/buildCSSForFlavoredJS.mjs

```diff
@@ -0,0 +1,11 @@
+export default function buildCSSForFlavoredJS(dynamicSizes) {
+  if (dynamicSizes == null || dynamicSizes.length === 0) return '';
+
+  return `<style>${Array.from(dynamicSizes, (charCount) =>
+    `@media(max-width:${charCount * 8 + 142 + 16 * 4 + (charCount > 68 ? 274 : 0)}px){` +
```

Comment:
> `8` is based on the size on `1ch` on `pre` element (which is `7.8025px`)

Is this measured or based on some standard? I'm just not sure if this value works for any mobile browser, especially when we don't specify any absolute value in css.

> `274` is the result of `234+32+24-16`, as you guessed because of the aside navigation menu for bigger screens.

This width is based on client's screen width and should be handled by css. I.e. logically we have:
```js
if (screenWidth > 1024) { // on our side we don't know value of screenWidth
  return charCount * 8 + 142 + 16 * 4 + 274; // we know charCount and constants
} else {
  return charCount * 8 + 142 + 16 * 4;
}
```
For example, if user's sceenWidth is `1000` and `charCount` is `75`, the returned value is `806` and it easily fits because there's no menu.

In css, maybe there is a more elegant way but it would look like this:
```css
@media (max-width: 1024px) {
  @media (max-width: ${value}px) {
    ${stylePatch}
    /* no fix between 806 and 1024 */
  }
}
@media (min-width: 1024px) {
  @media (max-width: ${value + 274}px) {
    ${stylePatch}
    /* fix between 1024 and 1080 */
  }
}
```

Reply:
> I don't see why mobile browsers would behave differently.

For example, on desktop Chrome we have `Settings -> Appearance -> Font size` which default to `Medium`. Selecting `Very Large` breaks this.
On mobile devices, the chances to meet "fonts-are-bigger-than-expected" situation are higher because of much higher DPI: users are more likely to adjust default settings, OS is more likely to provide custom zooming suggestions, particular browsers are more likely to enforce minimal readable size.

Overall I agree that this already solves the issue for most cases, and does it better than with any fixed value (although as a direct alternative we might consider not `600px` from #43495 but something like `1280px`). I was unsure if it can be improved further into exhaustive solution, or it must be kept as tradeoff.

> I'm all for improving the readability of the code, maybe by adding comments?

Yep, probably the whole `charCount * 8 + 142 + 16 * 4 + (charCount > 68 ? 274 : 0)` should be moved into function with brief comment/reference for each number, maybe even with `274` replaced by `234+32+24-16`.

Reply:
I've added a few comments that try to answer all the questions that have been raised in this review, let me know what you think.